app/lanes.go

```diff
@@ -0,0 +1,80 @@
+package app
+
+import (
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/client"
+
+	signerextraction "github.com/skip-mev/block-sdk/adapters/signer_extraction_adapter"
+	"github.com/skip-mev/block-sdk/block/base"
+	defaultlane "github.com/skip-mev/block-sdk/lanes/base"
+	mevlane "github.com/skip-mev/block-sdk/lanes/mev"
+)
+
+const (
+	maxTxPerMEVLane     = 500  // this is the maximum # of bids that will be held in the app-side in-memory mempool
+	maxTxPerDefaultLane = 3000 // all other txs
+)
+
+var (
+	defaultLaneBlockspacePercentage = math.LegacyMustNewDecFromStr("0.90")
+	mevLaneBlockspacePercentage     = math.LegacyMustNewDecFromStr("0.10")
+)
+
+// CreateLanes walks through the process of creating the lanes for the block sdk. In this function
+// we create three separate lanes - MEV, Free, and Default - and then return them.
+func CreateLanes(app *OsmosisApp, txConfig client.TxConfig) (*mevlane.MEVLane, *base.BaseLane) {
+	// 1. Create the signer extractor. This is used to extract the expected signers from
+	// a transaction. Each lane can have a different signer extractor if needed.
+	signerAdapter := signerextraction.NewDefaultAdapter()
+
+	// 2. Create the configurations for each lane. These configurations determine how many
+	// transactions the lane can store, the maximum block space the lane can consume, and
+	// the signer extractor used to extract the expected signers from a transaction.
+
+	// Create a mev configuration that accepts maxTxPerMEVLane transactions and consumes mevLaneBlockspacePercentage of the
+	// block space.
+	mevConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   mevLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerMEVLane,
+	}
+
+	// Create a default configuration that accepts maxTxPerDefaultLane transactions and consumes defaultLaneBlockspacePercentage of the
+	// block space.
+	defaultConfig := base.LaneConfig{
+		Logger:          app.Logger(),
+		TxEncoder:       txConfig.TxEncoder(),
+		TxDecoder:       txConfig.TxDecoder(),
+		MaxBlockSpace:   defaultLaneBlockspacePercentage,
+		SignerExtractor: signerAdapter,
+		MaxTxs:          maxTxPerDefaultLane,
+	}
+
+	// 3. Create the match handlers for each lane. These match handlers determine whether or not
+	// a transaction belongs in the lane.
+
+	// Create the final match handler for the mev lane.
+	factory := mevlane.NewDefaultAuctionFactory(txConfig.TxDecoder(), signerAdapter)
+	mevMatchHandler := factory.MatchHandler()
+
+	// Create the final match handler for the default lane. I.e this will direct all txs that are
+	// not free nor mev to this lane
+	defaultMatchHandler := base.DefaultMatchHandler()
+
+	// 4. Create the lanes.
+	mevLane := mevlane.NewMEVLane(
+		mevConfig,
+		factory,
+		mevMatchHandler,
+	)
```

Comment:
What do we think about adding the free lane logic and then commenting it out, so that way when the time comes we don't need to regain any context and we just uncomment it here?

Reply:
I think there is a few other things to be aware of when adding a free lane, and it would require commented out code in a few files, one being the ante handler. I feel it's best to gain full context on the feature when we decide to add it :thinking: 

wdyt?

Reply:
Yeah that SGTM